doc/source/reference/c-api/array.rst

```diff
@@ -74,6 +74,14 @@ and its sub-types).
     Clears the specified array flags. This function does no validation,
     and assumes that you know what you're doing.
 
+.. c:function:: void PyArray_HANDLER(PyArrayObject *arr)
+
+    .. versionadded:: 1.22
+
+    Returns a pointer to the memory handler of the array, which is
+    used for ``malloc``/``calloc``/``realloc``/``free`` per object.
+    Can return ``NULL`` for some cases.
+
```

Comment:
This is actually 
```
static inline NPY_RETURNS_BORROWED_REF PyObject *PyArray_HANDLER(PyArrayObject *arr);
```

It is similar to `PyDataMem_GetHandler`](https://numpy.org/doc/stable/reference/c-api/data_memory.html#c.PyDataMem_GetHandler) and should have almos the same doc string